File: Python/trainpassengers.py
capacity, stops = map(int, input().split(' '))
curr = 0
imp = False

for i in range(stops):
    left, entered, waited = map(int, input().split(' '))
    curr += entered - left

    if curr > capacity or curr < 0 or (capacity - curr > 0 and waited > 0):
        imp = True
        break

    if i == stops-1 and (waited > 0 or curr > 0) or i == 0 and left > 0:
        imp = True
        break

print("impossible" if imp else "possible")

File: Kotlin/trainpassengers.kt
fun main() {
    val (capacity, stops) = readLine()!!.split(' ').map(String::toInt)
    var curr = 0
    var imp = false

    for (i in 0..stops-1) {
        val (left, entered, waited) = readLine()!!.split(' ').map(String::toInt)
        curr += entered - left

        if (curr > capacity || curr < 0 || (capacity - curr > 0 && waited > 0)) {
            imp = true
            break
        }

        if (i == stops-1 && (waited > 0 || curr > 0) || i == 0 && left > 0) {
            imp = true
            break
        }
    }

    if (imp) println("impossible") else println("possible")
}